746 action chips

GCHQ-Developer-741 opened the pull request:
## Summary of the changes
Added a new `ic-action-chip` component, that combines the bahaviour of chip and button:
- Has 3 sizes, controlled by the `size` prop, which affect the padding.
- Has a `filled` & `outlined` variant, with a `transparent-background` prop that hides a filled background on just the outline variant.
- Can supply the relevant link and form props depending whether it is a button or a link. Setting an `href` on the component turns it into an `<a>` tag behind the scenes. Supplying a `target="_blank"` to the component will display the "Open in new window" icon next to the label.
- Can set the theme and monochrome to display different colours.
- Provided `icon` & `badge` slots to display extra information.
- Added the stories and tests for all the prop combinations.

## Related issue
#746 

## Checklist

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render.

ad9242 commented:
It might just be my eyes, but I don't see such as obvious difference as there appears to be in Figma when the hover & pressed states styling is applied to the outlined variant. In particular the color of the text\icons does not seem to lighten\darken as much as is shown in the designs:

Light mode (figma on top):

<img width="560" height="188" alt="image" src="https://github.com/user-attachments/assets/5487380b-53ab-423c-854a-0eac8fed6189" />


Dark mode:

<img width="615" height="196" alt="image" src="https://github.com/user-attachments/assets/cb9a3cf6-b969-4f5c-908b-a6bf503d8590" />

ad9242 commented:
Adding content to the `badge` or  `icon` slots after initial render does not happen: eg adding to `icon` slot on delay:

```
render: () => {
    const [iconRendered, setIconRendered] = React.useState(false);
    useEffect(() => {
      setTimeout(() => {
        setIconRendered(true);
      }, 3000);
    }, []);
  
    return (
      <IcActionChip label="Default">
        {iconRendered && (<svg
      slot="icon"
      width="20"
      height="20"
      viewBox="0 0 20 20"
      fill="currentColor"
      xmlns="http://www.w3.org/2000/svg"
      aria-label="account"
    >
      <path
        d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z"
      />
    </svg>)}
  </IcActionChip>)
  },
```

Also, when reversing the logic to remove the icon , a blank space is left:

<img width="190" height="83" alt="image" src="https://github.com/user-attachments/assets/224d8a77-e845-46e7-9d46-777511765e85" />

GCHQ-Developer-741 commented:
I've updated to watch for slot changes, added a cypress test for this behaviour also. For the colours, I'd missed the outlined hover and pressed colours for the icon and label, I've added them in. It will require an update to ic-button screenshots since I had to change a semantic icon colour (it was incorrect for button as well as this)
